src/meson.build

```diff
@@ -1,23 +1,22 @@
 # Compiled library
 
-args = ['-DTOML_HEADER_ONLY=0']
-
-if get_option('default_library') != 'static'
-	args += '-DTOML_SHARED_LIB=1'
-endif
+lib_args = []
+lib_args += universal_args
+# lib_args += ...
 
 lib = library(
 	meson.project_name(),
 	'toml++'/'toml.cpp',
-	cpp_args: args,
+	cpp_args: lib_args,
 	gnu_symbol_visibility: 'hidden',
 	include_directories: include_dir,
 	install: not is_subproject,
-	version: meson.project_version()
+	version: meson.project_version(),
+	override_options: overrides
 )
 
 tomlplusplus_dep = declare_dependency(
-	compile_args: args,
+	compile_args: lib_args,
```

Comment:
Uhm... Maybe this is not ideal, as it would force _everybody_ to compile their projects with the compile flags used to build the library

Reply:
I mean, not all flags are meant to be public, only the ones required to consume the library (i.e. things like `-DTOML_SHARED_LIB=1`). Exposing everything would not only be technically wrong, but could also cause compile failures for users (for example, if you compile the lib with Clang and a user consumes the pkg-config file and then compiles their project with GCC, the compilation would fail because it wouldn't be able to understand flags like `-Weverything`).

This is why I left them in `devel_args`, because while they are not _required_ to build the library, they are mostly interesting for you, the developer, so that you can take a look at all the warnings and fix them. But users shouldn't really care

Reply:
Example:

```text
$ CXX=clang++ meson setup build --prefix="$(pwd)/install" -Dcompile_library=true
$ meson install -C build
$ cat install/lib/x86_64-linux-gnu/pkgconfig/tomlplusplus.pc
prefix=/home/tachi/Scrivania/tomlplusplus/install
libdir=${prefix}/lib/x86_64-linux-gnu
includedir=${prefix}/include

Name: tomlplusplus
Description: TOML config file parser and serializer for C++
URL: https://marzer.github.io/tomlplusplus
Version: 3.0.1
Libs: -L${libdir} -ltomlplusplus
Cflags: -I${includedir} -ferror-limit=5 -Wno-unused-command-line-argument -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-documentation -Wno-documentation-unknown-command -Wno-switch-enum -Wno-covered-switch-default -DTOML_HEADER_ONLY=0 -DTOML_SHARED_LIB=1

```

Reply:
Oh, right. I guess I've misunderstood the relationship between the dependency flags and flags for just one target. I didn't realise it was passing the flags onto users consuming the library too.

Is there some way to apply the 'devel' flags to the lib when it is built, but not pass them on to the users? I'd still like the optional pedantic warning setting to apply if I choose it.